src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,843 @@
+		}
+
+		$data['attachment'] = []; // @ToDo
+		$data['tag'] = self::createTagList($item);
+		$data = array_merge($data, self::createPermissionBlockForItem($item));
+
+		return $data;
+	}
+
+	/**
+	 * @brief Transmits a profile deletion to a given inbox
+	 *
+	 * @param integer $uid User ID
+	 * @param string $inbox Target inbox
+	 */
+	public static function transmitProfileDeletion($uid, $inbox)
+	{
+		$owner = User::getOwnerDataById($uid);
+		$profile = APContact::getByURL($owner['url']);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => 'Delete',
+			'actor' => $owner['url'],
+			'object' => $owner['url'],
+			'published' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+			'to' => [ActivityPub::PUBLIC_COLLECTION],
+			'cc' => []];
+
+		$signed = LDSignature::sign($data, $owner);
+
+		logger('Deliver profile deletion for user ' . $uid . ' to ' . $inbox .' via ActivityPub', LOGGER_DEBUG);
+		HTTPSignature::transmit($signed, $inbox, $uid);
+	}
+
+	/**
+	 * @brief Transmits a profile change to a given inbox
+	 *
+	 * @param integer $uid User ID
+	 * @param string $inbox Target inbox
+	 */
+	public static function transmitProfileUpdate($uid, $inbox)
+	{
+		$owner = User::getOwnerDataById($uid);
+		$profile = APContact::getByURL($owner['url']);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => 'Update',
+			'actor' => $owner['url'],
+			'object' => self::profile($uid),
+			'published' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+			'to' => [$profile['followers']],
+			'cc' => []];
+
+		$signed = LDSignature::sign($data, $owner);
+
+		logger('Deliver profile update for user ' . $uid . ' to ' . $inbox .' via ActivityPub', LOGGER_DEBUG);
+		HTTPSignature::transmit($signed, $inbox, $uid);
+	}
+
+	/**
+	 * @brief Transmits a given activity to a target
+	 *
+	 * @param array $activity
+	 * @param string $target Target profile
+	 * @param integer $uid User ID
+	 */
+	public static function transmitActivity($activity, $target, $uid)
+	{
+		$profile = APContact::getByURL($target);
+
+		$owner = User::getOwnerDataById($uid);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => $activity,
+			'actor' => $owner['url'],
+			'object' => $profile['url'],
+			'to' => $profile['url']];
+
+		logger('Sending activity ' . $activity . ' to ' . $target . ' for user ' . $uid, LOGGER_DEBUG);
+
+		$signed = LDSignature::sign($data, $owner);
+		HTTPSignature::transmit($signed, $profile['inbox'], $uid);
+	}
+
+	/**
+	 * @brief Transmit a message that the contact request had been accepted
+	 *
+	 * @param string $target Target profile
+	 * @param $id
+	 * @param integer $uid User ID
+	 */
+	public static function transmitContactAccept($target, $id, $uid)
```

Comment:
Please rename to `sendContactAccept` to avoid repeating "transmit" when calling the method.